src/util/adb.js

```diff
@@ -271,6 +272,17 @@ export default class ADBUtils {
       value: `-profile ${deviceProfileDir}`,
     }];
 
+    for (const browser of packageIdentifiers) {
+      if (apk.startsWith(browser) && apkComponent) {
+        if (apk !== browser && !apkComponent.includes('.')) {
```

Comment:
This logic should be after the following block below, to make sure that the `browser` is also prepended by default (`.App`):
```
    if (!apkComponent) {
      apkComponent = '.App';
    } else if (!apkComponent.includes('.')) {
      apkComponent = `.${apkComponent}`;
    }
```

It would be nice to have unit tests to cover these cases.

And then the logic should be like this:

```
if (apkComponent.startsWith(".")) {
  // loop over packageIdentifiers,
  // and if apk is either identical to the package,
  // or apk starts with `${package}.`,
  // prepend the package identifier before `apkComponent`
  // and break. 
}
```

Reply:
> or apk starts with `${package}.`

Ah! That's a good idea for the whole "geckoview" vs "geckoview_example"